enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+(ARF)](https://csrc.nist.gov/Projects/Security-Content-Automation-Protocol/Specifications/arf).
+These reports can be large and verbose. A single report for a small cluster can
+generate about 3 MB of data. This increase with each scan run, and with the
+amount of infrastructure being scanned.
+
+Historically, the Compliance Operator saves these reports to a physical volume.
+This works around saving the entire report in Custom Resource Definitions,
+which are bound to limitations of etcd.
+
+Unfortunately, physical volumes have their own short-comings. Reports stored in
+physical volumes still need to be fetched for evidence. Physical volumes can
+also be deleted like any other cluster resource.
+
+A improvement to the idea of putting evidence reports on disk would be to dump
+them into an object or file store. This has an added benefit of allowing
+administrators managing multiple clusters to put evidence in a single location.
+
+## Motivation
+
+The motivation of the enhancement is to store compliance evidence outside the
+cluster and its storage pools.
+
+### Goals
+
+The goal is to provide a way for Kubernetes administrators to configure the
+Compliance Operator to send compliance reports, specifically ARF reports, to an
+external storage pool, like an object or file store.
+
+### Non-Goals
+
+It is not the goal of this enhancement to implement results forwarding, or
+adjust the definition of evidence outside an ARF report.
+
+## Proposal
+
+Provide an additional configuration for scan settings that allows site
+reliability engineers to forward evidence to a storage pool. Today, the
+Compliance Operator provides a `default` `ScanSetting` that integrates with
+persistent storage:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could create another `ScanSetting` that bypasses
+storing evidence on physical volumes with the following:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage
+  namespace: openshift-compliance
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could configure both storage mechanisms for redundancy:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage-and-cluster-storage
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
```

Comment:
I am wondering how we would distinguish what type of storage it is here? ex. S3 or Ceph

Reply:
Yeah - maybe we could do something like:

```
externalResultStorage:
  s3:
    endpoint: ...
```

Reply:
I like this option the most. At first I was going to suggest:
```
externalResultStorage:
    endpoint: ...
    type: S3
```
But I like your option better because we could potentially configure more for redundancy if we choose to, we could also implement a `none` storage type with no nested attributes. Also, the Go implementation would be nicer because each `nested` struct could keep its own separate attributes as opposed to lumping them all together,

Reply:
I think specifying a type would be ideal (there can be reasonable defaults if needed). Ceph has several endpoints for storing data, one of them being Ceph RADOS Gateway (or Ceph Object Gateway) which does support S3 as a protocol. It also supports OpenStack's Swift (in case y'all need that). By supporting configurable protocols you ensure portability. `s3` was initially suggested because it's supported by AWS's S3, by Ceph radosgw, MinIO and other object storage implementations. So it would give the most value as an initial implementation.